connection-cache/src/connection_cache.rs

```diff
@@ -38,6 +38,9 @@ pub trait ConnectionManager: Send + Sync + 'static {
 
     fn new_connection_pool(&self) -> Self::ConnectionPool;
     fn new_connection_config(&self) -> Self::NewConnectionConfig;
+    fn update_key(&mut self, _key: &Keypair) -> Result<(), Box<dyn std::error::Error>> {
+        Ok(())
+    }
```

Comment:
This looks like a bad default implementation, and I am not sure we need a default implementation here.
Can you please remove it if not needed?